CMS for new Support Pages - Initial models and forms

ldodds opened the pull request:
Adds a basic set of CMS classes for managing content for the new support help pages:

- Category
- Page, which belongs to a Category
- Section, which belongs to a Page and which have an order

The PR includes a basic UI for displaying this content to users, although this is still largely a mockup and needs a redesign. The tests for this don't do anything more than confirm that pages render

The rest of the PR is primarily focused on the core admin interface for managing that content. There are forms for creating and updating all of the models, along with quick links to publish, hide and preview content, add pages to categories, add sections to pages. The page form allows reordering of Sections.

The models are translation ready, but have not yet added the code for synchronising to Transifex.

Other things to come in future PRs:

- better specs for the public interface, which needs proper design and components for parts of the page
- extracting some common logic in the admin interface, e.g. the publish/hide buttons
- more tests, e.g. around reordering of sections
- adding links to edit and manage content from the public view, for admins

urbanwide commented:
![image](https://github.com/user-attachments/assets/1ea83bcd-99ab-4708-8d85-b834db0ea5e6)

I think you could do with putting a 'Page' label or similar above the page dropdown on the section admin interface?
I wonder if when you are editing a page, you could also do with Category and Audience labels too?

urbanwide commented:
Although now I have read your description at the top, you may end up coming back to these things :-)

urbanwide commented:
![image](https://github.com/user-attachments/assets/bda099a5-29b0-410e-a825-7493900d5719)
I'm clicking save after adding sections here, and it's either saving and not redirecting afterwards, or not saving and not displaying an error. Everything looks ok...(no category or audience missing)

ldodds commented on config/routes.rb:
```diff
@@ -534,6 +541,27 @@
+    namespace :cms do
+      resources :categories, except: [:show] do
+        member do
+          put :publish
```
Added. I'd concerned a concern in the controller, but not the routing.

ldodds commented on spec/system/admin/cms/manage_categories_spec.rb:
```diff
@@ -0,0 +1,71 @@
+        fill_in 'Description en', with: 'Category Description'
+        fill_in 'Icon', with: 'lightbulb'
+
+        expect { click_on 'Save' }.to change(Cms::Category, :count).by(1)
```
Moved some of this around as suggested. Although I tend to think of filling in fields in a system spec as part of the test rather than the setup. Mostly because you're testing that the field exists and can be filled in as well as what happens afterwards. But I guess the only difference is where the error might be triggered if a form isn't complete.

ldodds commented:
Could you give this another review please @urbanwide. Added form labels, added error partial to form, concern and tidied the specs a bit. Will do rest in future PRs.

The workflow around page/section creation probably needs a bit of work. I was also considering moving the section management to a separate form.

urbanwide reviewed (approved):
Looks good 👍🏻
